### PR TITLE
Added sample code for ArgumentCustomization

### LINKED ISSUE #758: Better document ToolSettings ArgumentCustomization property
ArgumentCustomization  need better help and sample code, to make it's purpose clear and how it's used.

ArgumentCustomization is in general the exception and not the rule, as it's used when Cake alias / settings don't provide what's necessary, i.e. when a tool adds an parameter before Cake knows about it.

Code:
* https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Tooling/ToolSettings.cs#L21 

Documentation:
* http://cakebuild.net/api/cake.core.tooling/d75cd243/60ae911b

### PR DESCRIPTION
**Pull request by devlead**
This fixes #758

### CODE REVIEW

File: src/Cake.Core/Tooling/ToolSettings.cs [hunk 1]
```diff
@@ -16,8 +16,18 @@ public class ToolSettings
...
-        /// <value>The arguments.</value>
+        /// <example>
+        /// <code>
+        /// NuGetAddSource("Cake", "https://www.myget.org/F/cake/api/v3/index.json",
+        ///     new NuGetSourcesSettings { UserName = "user", Password = "incorrect",
+        ///     ArgumentCustomization = args=&gt;args.Append("-StorePasswordInClearText")
+        /// });
+        /// </code>
+        /// </example>
+        /// <value>The arguments Func.</value>
```
Comment (patriksvensson) [inline src/Cake.Core/Tooling/ToolSettings.cs]: Suggestion: `The delegate used to customize the <see cref="Cake.Core.IO.ProcessArgumentBuilder" />.`.

Comment (devlead) [inline src/Cake.Core/Tooling/ToolSettings.cs]: Agree, I'll fix.

**Comment by patriksvensson** [comment]
@devlead Can you rebase this against develop and I'll merge it.

**Comment by devlead** [comment]
@patriksvensson rebased.

**Pull request merged**